### PR TITLE
Create daysUntil.md

### PR DESCRIPTION
**Pull request by AlstrupJacob**
also edit tags documentation to included required tags. solves #3811 

### Description

created daysUntil.md  in docs/content/kotlin/concepts/datetime/terms/daysUntil and edited tags documentation to include required tags

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Kotlin'
+  - 'Instant'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.daysUntil()`** method returns an Int of whole days between two instants. if the number is too large to fit in an Int datatype, the method will return Int.MAX_VALUE or Int.MIN_VALUE depending on the sign of the result.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:15]:
```suggestion
The **`.daysUntil()`** method calculates the number of whole days between two instants and returns the result as an `Int`. If the number of days is too large to fit within the bounds of the `Int` datatype, the method will return `Int.MAX_VALUE` if the result is positive or `Int.MIN_VALUE` if the result is negative.
```

File: content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Syntax
+
+```pseudo
+Instant.daysUntil(Instant)
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:20]:
```suggestion
fun Instant.daysUntil(other: Instant, timeZone: TimeZone): Int
```

File: content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+-`Instant`: A moment in time.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:23]:
```suggestion
-`Instant`: A moment in time.
- `TimeZone`: This parameter is of type `TimeZone`. It indicates that the calculation should consider the time zone when determining the number of days between the two `Instant` objects.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes from my side

File: content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example
+
+The example demonstrates the use of `.daysUntil()` to calculate the number of days between two Instants.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:27]:
```suggestion
The example demonstrates the use of `.daysUntil()` to calculate the number of days between two `Instants`.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]

File: content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+The example demonstrates the use of `.daysUntil()` to calculate the number of days between two `Instants`.
+
+```kotlin
+fun main() {
+  Instant t1 = Instant.now()
+
+  Instant t2 = t1.plus(125, HOURS)
+
+  Instant t3 = t2.daysUntil(t1)
+  println(t3)
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:37]: Can you show me a screenshot of where this is working and give the output as mentioned below?

Comment (dakshdeepHERE) [reply content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:37]: I think we need to use `ChronoUnit` class to make it work.

**Comment by dakshdeepHERE** [comment]
@AlstrupJacob please read my recent comment and try to fix it so we can move forward to 2nd review.

**Comment by AlstrupJacob** [comment]
> @AlstrupJacob please read my recent comment and try to fix it so we can move forward to 2nd review.

I think it works now, I'm having trouble importing the libraries required

**Comment by dakshdeepHERE** [comment]
> > @AlstrupJacob please read my recent comment and try to fix it so we can move forward to 2nd review.
> 
> I think it works now, I'm having trouble importing the libraries required

I can't see it reflected here on git. Have you pushed the commits?

**Comment by AlstrupJacob** [comment]
> > > @AlstrupJacob please read my recent comment and try to fix it so we can move forward to 2nd review.
> > 
> > 
> > I think it works now, I'm having trouble importing the libraries required
> 
> I can't see it reflected here on git. Have you pushed the commits?

yes push 47ae83d

File: content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+```kotlin
+fun main() {
+  val t1 = Instant.parse("2008-09-29T12:00:00Z")
+
+  val t2 = t1.plus(125, DateTimeUnit.HOURS)
+
+  val t3 = t2.daysUntil(t1)
+  println(t3)
+}
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/daysUntil/daysUntil.md:38]:
```suggestion
import java.time.Instant
import java.time.Duration

fun Instant.daysUntil(other: Instant): Long {
// Calculate the duration between 'this' Instant and 'other' Instant
    val duration = Duration.between(this, other)
    
// Convert the duration to days and return the result 
    return duration.toDays()
}

fun main() {
    val t1 = Instant.parse("2008-09-29T12:00:00Z")
    val t2 = t1.plus(Duration.ofHours(125))

    val t3 = t2.daysUntil(t1)
    println(t3)
}
```
There are two big problems here:
- `DateTimeUnit.HOURS` is not a valid enum value in Kotlin or the standard Java library. You should use `Duration.ofHours(125)` to add `125` hours to `t1.`
- The `daysUntil` function does not exist in the standard Kotlin or Java libraries for Instant objects. You need to create a custom extension function for Instant to calculate the days between two Instant timestamps, as I showed in the previous response.
I have added the fixed code please commit it.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Last change

**Comment by dakshdeepHERE** [comment]
@AlstrupJacob feel free to ask anything. I have last commit request from my side.

**Comment by dakshdeepHERE** [comment]
@AlstrupJacob Looks good for second review.

**Review by dakshdeepHERE — APPROVED** [review]

**Review by ishg-153 — APPROVED** [review]
This looks good to merge.

**Comment by dakshdeepHERE** [comment]
@ishg-153 I'm merging this as you passed it in the second review. Thanks for the review. 🚀🛠️

**Pull request merged**